app/src/main/java/pl/szczodrzynski/edziennik/data/api/edziennik/librus/data/api/LibrusApiTeacherFreeDays.kt

```diff
@@ -36,12 +36,13 @@ class LibrusApiTeacherFreeDays(override val data: DataLibrus,
                 val id = teacherAbsence.getLong("Id") ?: return@forEach
                 val teacherId = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
                         ?: return@forEach
-                val type = teacherAbsence.getJsonObject("Type").getLong("Id") ?: return@forEach
+                val type = teacherAbsence.getJsonObject("Teacher")?.getLong("Id")
+                    ?: return@forEach
```

Comment:
What is this?

Reply:
Type object is not returned in JSON anymore. App could not display teacher absence when any variable was assigned with non existing object. I could not find way to not declarete the data (leak of knowedge and expirence) so I used this one. Yes it's dirty. What is better solution?

Reply:
The app expects a Type ID in this field. Assigning a Teacher ID does not make it find a "valid" object, because Teacher is not a Type. Try putting -1 as the type ID to see if it works. If not, you have to modify the logic of displaying the absence list in the UI.